src/sonic-yang-models/yang-models/sonic-system-port.yang

```diff
@@ -0,0 +1,81 @@
+module sonic-system-port {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/Azure/sonic-system-port";
+    prefix system-port;
+
+    import sonic-port {
+        prefix port;
+    }
+
+    import sonic-types {
+        prefix stypes;
+    }
+
+    description "SYSTEM_PORT YANG Module for SONiC OS";
+
+    revision 2022-11-10 {
+        description "First Revision";
+    }
+
+    container sonic-system-port {
+
+        container SYSTEM_PORT {
+
+            description "SYSTEM_PORT table of config_db.json";
+
+            list SYSTEM_PORT_LIST {
+
+                key "hostname asic_name ifname";
+
+                leaf hostname {
+                    type stypes:hostname;
+                }
+
+                leaf asic_name {
+                    type string;
+                }
+
+                leaf ifname {
+                    type string;
+                }
+
+                leaf core_index {
+                    type uint8;
```

Comment:
Allowed 0-255 ?

Reply:
8-bits give us flexibility to support more cores in future ASICs. Admittedly, 255 cores is not likely. If you prefer, I can constrain to maximum 8 cores as so:

type uint8 {
   range 0..7;
}